Userland/Libraries/LibCore/GamepadFinderLinux.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2024, Undefine <undefine@undefine.pl>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Bitmap.h>
+#include <AK/Platform.h>
+#include <AK/String.h>
+#include <LibCore/GamepadFinder.h>
+#include <LibCore/System.h>
+
+#include <libudev.h>
+#include <linux/input.h>
+
+#if !defined(AK_OS_LINUX)
+static_assert(false, "This file must only be used for Linux");
+#endif
+
+namespace Core {
+
+static ErrorOr<bool> is_gamepad(StringView path)
+{
+    if (auto result = System::open(path, O_RDONLY); !result.is_error()) {
```

Comment:
```suggestion
    auto result_or_error = System::open(path, O_RDONLY);
    if (result_or_error.is_error()) {
        warnln("Failed to open input device {}: {}", path, result.release_error());
        return false;
    }
    
    int fd = result_or_error.release_value();
    ...
```

Reduces nesting and makes it a little easier to follow